File: OOP/03_05/staticdata/java/NumberAppStatic/src/main/java/edu/brooklyn/NumberUtil.java
package edu.brooklyn;

public class NumberUtil {

	public int sumToNumberWithAuto(int number) {
		int sum = 0;
		for (int i = 0; i < number; i++) {
			sum += i;
		}
		return sum;
	}

	public int sumToNumberWithStatic(int number) {
		for (int i = 0; i < number; i++) {
			sum += i;
		}
		return sum;
	}

	private static int sum = 0;
}

File: OOP/03_05/staticdata/cpp/numberutil.cc
#include "numberutil.h"

namespace Edu {
    namespace Brooklyn {

        int NumberUtil::sumToNumberWithAuto(int number) const {
            int sum = 0;
            for (int i=0; i<=number; i++) {
                sum += i;
            }
            return sum;
        }

        int NumberUtil::sumToNumberWithStatic(int number) const {
            static int sum = 0;
            for (int i=0; i<=number; i++) {
                sum += i;
            }
            return sum;
        }

    }

}

